### PR TITLE
Lime 1248

### PR DESCRIPTION
**Pull request by RachelRobGDS**
### What changed

SA payment details userInput amended - key names amended and 'amount' value converted to number from string.

### Why did it change

To fit HMRC requirements re data type and key names of this data.

### Issue tracking

- [LIME-1248](https://govukverify.atlassian.net/browse/LIME-1248)

Related PR has already been merged for the [third-party-stub](https://github.com/govuk-one-login/ipv-third-party-stubs/pull/238) to remove a troublesome space from in front of the amount for user AL000000S to ensure third-party-stub test users are now in keeping with the format expected by HMRC. This has been confirmed by testing user journeys with the minor third-party-stub change and this FE change.

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

[LIME-1248]: https://govukverify.atlassian.net/browse/LIME-1248?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ElliotMurphyGDS** [comment]
Can we fix code smell to get passing action?

File: tests/unit/src/app/kbv/controllers/self-assessment-payment-question.test.js [hunk 1]
```diff
@@ -43,8 +43,12 @@ describe("self-assessment-payment-question controller", () => {
         req.session.question.questionKey = questionKey;
         req.form.values = {
           selfAssessmentPaymentDate: "2023-02-22",
-          selfAssessmentPaymentAmount: 2000.22,
+          selfAssessmentPaymentAmount: "2000.22",
         };
+        const userInput = JSON.stringify({
+          amount: new Number(req.form.values.selfAssessmentPaymentAmount),
```
Comment (smsgds) [inline tests/unit/src/app/kbv/controllers/self-assessment-payment-question.test.js:49]: Can you try 
`amount: parsFloat(req.form.values.selfAssessmentPaymentAmount),`

and see if it behaves as expected

**Review by smsgds — CHANGES_REQUESTED** [review]

File: tests/unit/src/app/kbv/controllers/self-assessment-payment-question.test.js (hunk 1 above)
Comment (RachelRobGDS) [reply tests/unit/src/app/kbv/controllers/self-assessment-payment-question.test.js]: Did as above and it works

**Comment by RachelRobGDS** [comment]
Code smell fixed as per Seamus' suggestion. Will squash commits following review.

**Pull request closed without merging**